src/Core/Grid/Column/Column.php

```diff
@@ -0,0 +1,275 @@
+<?php
+/**
+ * 2007-2018 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Grid\Column;
+
+use InvalidArgumentException;
+use PrestaShop\PrestaShop\Core\Grid\Exception\InvalidColumnDataException;
+use Symfony\Component\Form\FormTypeInterface;
+
+/**
+ * Class Column is responsible for defining single column in row
+ */
+final class Column implements ColumnInterface
+{
+    /**
+     * @var string
+     */
+    private $type;
+
+    /**
+     * @var string Translated column name
+     */
+    private $name;
+
+    /**
+     * @var string  Unique column identifier
+     */
+    private $id;
+
+    /**
+     * @var callable|null Column's content modifier if needed
+     */
+    private $modifier;
+
+    /**
+     * @var string|null Fully qualified class name of Symfony's form type if column is filterable
+     */
+    private $filterFormType;
+
+    /**
+     * @var array Form type options for $filterFormType
+     */
+    private $filterFormTypeOptions = [];
+
+    /**
+     * @var bool True if column is sortable or False otherwise
+     */
+    private $isSortable = true;
+
+    /**
+     * @var int Column's position in grid
+     */
+    private $position = 0;
+
+    /**
+     * @var array
+     */
+    private $options;
```

Comment:
options are the column extra attributes, am I right?

Reply:
yes, at the moment i kept it as plain `array`, but i guess we need an interface for that, right?